mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala

```diff
@@ -132,7 +132,7 @@ class BisectingKMeansModel private[ml] (
   private var trainingSummary: Option[BisectingKMeansSummary] = None
 
   private[clustering] def setSummary(summary: BisectingKMeansSummary): this.type = {
-    this.trainingSummary = Some(summary)
+    this.trainingSummary = Option(summary)
```

Comment:
per @holdenk's suggestion, I changed the `setSummary` to use `Option.apply` which treats `null` as `None`. This allows us to exercise the test case for `hasSummary` when `summary == None`. This was never tested in Scala either so I added unit tests for both Scala and Python.